apps/zotonic_core/src/db/z_db.erl

```diff
@@ -874,20 +874,24 @@ equery(Sql, Parameters, Context, Timeout) ->
 
 %% @doc Execute the same SQL statement for a list of parameters. Default timeout
 %% of 30 seconds.
--spec execute_batch(SQL, ParametersList, Context) -> query_result() when
+-spec execute_batch(SQL, ParametersList, Context) -> Result when
     SQL :: sql(),
     ParametersList :: list( parameters() ),
-    Context :: z:context().
+    Context :: z:context(),
+    Result :: {ok, [ query_result() ]}
+            | {error, term()}.
```

Comment:
I find this a bit odd, I see that in `z_db_pgsql.erl` these types are defined as:
```erlang
-type query_result() :: squery_result() | equery_result().

-type squery_result() :: epgsql_cmd_squery:response()
                       | epgsql_sock:error()
                       | {error, connection_down | term()}.

-type equery_result() :: epgsql_cmd_equery:response()
                       | epgsql_sock:error()
                       | {error, connection_down | term()}.
```
which means that this function could return something like: `{ok, [{error, connection_down]}`.

Is this intentional? Shouldn't it return `{error, connection_down}` in such cases instead?

Reply:
The types are a bit messy, especially for the batches.
The individual results could indeed have an error.
And overall we could have an error submitting the batch.

So we could end up with something like `{ok, [ {error, ...} ]}`
At least we _could_ submit the batch, but the first row of the batch gave an error.